test/commandbuilderharness.js

```diff
@@ -0,0 +1,161 @@
+const { expect } = require("chai");
+const { ethers } = require("hardhat");
+const weiroll = require("@weiroll/weiroll.js");
+
+async function deployLibrary(name) {
+  const factory = await ethers.getContractFactory(name);
+  const contract = await factory.deploy();
+  return weiroll.Contract.fromEthersContract(contract);
+}
+
+describe("CommandBuilderHarness", function () {
+  let cbh;
+  let math;
+  let strings;
+  let abi = ethers.utils.defaultAbiCoder;
+
+  before(async () => {
+    const Cbh = await ethers.getContractFactory("CommandBuilderHarness");
+    cbh = await Cbh.deploy();
+
+    math = await deployLibrary("Math");
+    strings = await deployLibrary("Strings");
+  });
+
+  async function executeBuildInputs(commands, state, abiout, msg){
+    for (let c of commands) {
+        selector = c.slice(0, 10);
```

Comment:
Ethers has a `hexDataSlice` that may make these easier to write and reason about.